pkg/generator/specification.go

```diff
@@ -50,13 +50,18 @@ func NewSpecificationGenerator(seed int64) *SpecificationGenerator {
 // generateIATPerGranularity generates IAT for one minute based on given number of invocations and the given distribution
 func (s *SpecificationGenerator) generateIATPerGranularity(numberOfInvocations int, iatDistribution common.IatDistribution, shiftIAT bool, granularity common.TraceGranularity) ([]float64, float64) {
 	if numberOfInvocations == 0 {
+		// no invocations in the current minute
 		return []float64{}, 0.0
 	}
 
 	var iatResult []float64
 	totalDuration := 0.0 // total non-scaled duration
 
-	for i := 0; i < numberOfInvocations; i++ {
+	// first invocation at the beginning of minute
+	iatResult = []float64{0.0}
+	endIndex := numberOfInvocations - 1
```

Comment:
Because of that, non-equidistant would break: with two invocations, it would always return [0, 60], while the invocation should be somewhere in the middle.

Reply:
Fixed in https://github.com/vhive-serverless/invitro/pull/557/commits/134b82c6b3a21b974bf84c914b2668cfe2968ff8.